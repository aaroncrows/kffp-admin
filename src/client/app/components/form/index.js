import React, { Component } from 'react';
import { connect } from 'react-redux';
import cuid from 'cuid';
import FormFields from './fields';
import RaisedButton from 'material-ui/RaisedButton';
import { prepareFormSubmit } from '../../actions/formActions';

const mapStateToProps = (state) => {
    return {
        form: state.form,
        model: state.model
    }
};

class Form extends Component {
    constructor (props) {
        super(props);

        this.renderField = this.renderField.bind(this);
        this.renderFormFields = this.renderFormFields.bind(this);
        this.submitHandler = this.submitHandler.bind(this);
        this.renderErrors = this.renderErrors.bind(this);
    }

    renderField (fieldData, fieldName) {
        const { dispatch } = this.props;

        const { fieldType, hintText, label, name, items, value, searchResults } = fieldData;
        const FormField = FormFields[fieldType];

        if (FormField) {
            return (
                <FormField
                    label={label}
                    hintText={hintText}
                    name={name}
                    id={`${name}_field`}
                    value={value}
                    fieldName={fieldName}
                    items={items}
                    searchResults={searchResults}
                    dispatch={dispatch}
                />
            );
        } else {
            return (
                <div>No matching field.</div>
            );
        }
    }

    renderFormFields (fields) {
        return Object.keys(fields).map((field, i) => {
            return (
                <div key={i}>
                    {this.renderField(fields[field], field)}
                </div>
            )
        });
    }

    submitHandler (e) {
        e.preventDefault();
        // TODO perform validation
        this.props.dispatch(prepareFormSubmit());
    }

    renderErrors (errors) {
        if (errors.length) {
            return errors.map((err, i) => {
                console.log(err);
                return (
                    <p key={i}>{err}</p>
                )
            });
        }
    }

    render () {
        const { fields, modelName, formType, data, errors } = this.props.form;
        const formTitle = `${formType} ${modelName} form`;

        return (
            <div>
                <h1>{formTitle}</h1>
                <div>
                    {this.renderErrors(errors)}
                </div>
                <form onSubmit={this.submitHandler}>
                    {this.renderFormFields(fields, data)}
                    <RaisedButton
                        label="submit"
                        primary={true}
                        type="submit"
                    />
                </form>
            </div>
        );
    }
};

export default connect(mapStateToProps)(Form);
