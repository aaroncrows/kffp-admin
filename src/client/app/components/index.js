import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Search from './search';
import { SearchResults } from './searchResults';

const mapStateToProps = (state) => {
    return {
        searchResults: state.search.searchResults
    };
};

class App extends Component {
    render() {
        const { dispatch, searchResults } = this.props;

        return (
            <div>
                <Search dispatch={dispatch} />
                <SearchResults
                    dispatch={dispatch}
                    searchResults={searchResults}
                />
            </div>
        )
    }
};

App.propTypes = {
    dispatch: PropTypes.func.isRequired
};

export default connect(mapStateToProps)(App);
