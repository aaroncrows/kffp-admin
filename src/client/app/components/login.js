import React, { Component, PropTypes } from 'react';
import RaisedButton from 'material-ui/RaisedButton';

class Login extends Component {
    constructor (props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick () {
        const { email, password } = this;
        const credentials = {
            email: email.value.trim(),
            password: password.value.trim()
        };

        this.props.onLoginClick(credentials);
    }

    render () {
        const { errorMessage } = this.props;

        return (
            <div>
                <input type="text" ref={(c) => { this.email = c; }} placeholder="Email" />
                <input type="password" ref={(p) => { this.password = p; }} placeholder="Password" />
                <RaisedButton onClick={e => this.handleClick(e)} label="Login" />

                {errorMessage &&
                    <p>{errorMessage}</p>
                }
            </div>
        );
    }
}

Login.propTypes = {
    onLoginClick: PropTypes.func.isRequired,
    errorMessage: PropTypes.string
};

export default Login;
