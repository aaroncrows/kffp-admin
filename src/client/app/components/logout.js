import React, { PropTypes } from 'react';

const Logout = ({ onLogoutClick }) => {
    return (
        <button onClick={() => onLogoutClick()}>Logout</button>
    );
};

Logout.propTypes = {
    onLogoutClick: PropTypes.func.isRequired
};

export { Logout };
