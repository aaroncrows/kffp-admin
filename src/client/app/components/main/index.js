import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getUserShows } from '../../actions/showActions';
import ShowSelect from './select';

const mapStateToProps = state => ({
    auth: state.auth,
    modal: state.modal,
    show: state.show
});

class Main extends Component {
    componentWillReceiveProps (nextProps) {
        const { auth } = nextProps;
        const { user } = auth;

        if (this.props.auth.user !== user) {
            const { displayName } = user;

            this.props.dispatch(getUserShows(displayName));
        }
    }

    render () {
        const { show, dispatch } = this.props;
        const { shows } = show;

        return (
            <div>
                <h1>Select a show</h1>
                <ShowSelect dispatch={dispatch} shows={shows} />
                
            </div>
        );
    }
}

export default connect(mapStateToProps)(Main);
