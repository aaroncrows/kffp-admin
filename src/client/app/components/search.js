import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux';
import { searchInput, searchForm } from '../actions/searchActions';

const mapStateToProps = (state) => {
    return {
        currentSearch: state.search.currentSearch
    };
};

export class Search extends Component {
    constructor (props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange (e) {
        this.props.dispatch(searchInput(e.target.value));
    }

    handleSubmit (e) {
        const { currentSearch } = this.props;
        e.preventDefault();
        this.props.dispatch(searchForm(currentSearch));
    }

    render () {
        const { currentSearch } = this.props;
        const errorMessage = false;

        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <div className='mdl-textfield mdl-js-textfield'>
                        <input type='text' value={currentSearch} onChange={this.handleChange} className='mdl-textfield__input' id='search-input'/>
                        <label className='mdl-textfield__label' htmlFor='search-input'>Enter Search Query</label>
                    </div>
                    <button type='submit' className='mdl-button mdl-js-button mdl-button--raised mdl-js-ripple-effect mdl-button--accent'>Search</button>
                </form>

                {errorMessage &&
                  <p>{errorMessage}</p>
                }
            </div>
        );
    }
}

Search.propTypes = {
    search: PropTypes.object,
    dispatch: PropTypes.func.isRequired
};

export default connect(mapStateToProps)(Search);
