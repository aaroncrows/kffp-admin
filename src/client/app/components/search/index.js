import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux';
import TextField from 'material-ui/TextField';
import { searchInput, searchForm } from '../../actions/searchActions';

const mapStateToProps = (state) => {
    return {
        search: state.search
    };
};

export class Search extends Component {
    constructor (props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange (e) {
        this.props.dispatch(searchInput(e.target.value));
    }

    handleSubmit (e) {
        const { currentSearch } = this.props.search;

        e.preventDefault();
        this.props.dispatch(searchForm(currentSearch));
    }

    render () {
        const { currentSearch } = this.props.search;
        const errorMessage = false;

        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <div>
                        <TextField
                            onChange={this.handleChange}
                            value={currentSearch}
                            floatingLabelText={'Enter a search query'}
                            hintText={'Enter a search query'}
                        />
                    </div>
                </form>

                {errorMessage &&
                  <p>{errorMessage}</p>
                }
            </div>
        );
    }
}

Search.propTypes = {
    search: PropTypes.object,
    dispatch: PropTypes.func.isRequired
};

export default connect(mapStateToProps)(Search);
