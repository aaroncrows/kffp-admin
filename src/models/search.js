import Boom from 'boom';

const userSearchHandler = (request, reply) => {
    const { db } = request.server.plugins.mongodb;
    const { text } = request.query;

    db.collection('users').find({ email: { $regex: `${text}`, $options: '$i' } },
        { password: 0 }, async (err, cursor) => {
            if (err) {
                return reply(Boom.internal('internal mongodb error', err));
            }

            const users = await cursor.toArray();

            return reply(users);
        }
    );
};

export default userSearchHandler;
