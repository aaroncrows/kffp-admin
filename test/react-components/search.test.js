import React from 'react';
import renderer from 'react-test-renderer';
import { Search } from '../../src/client/app/components/search';

describe('components', () => {
    describe('<Search />', () => {
        it('renders correctly', () => {
            const tree = renderer.create(<Search dispatch={() => true} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
