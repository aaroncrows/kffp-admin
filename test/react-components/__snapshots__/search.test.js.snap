exports[`components <Search /> renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <div
      className="mdl-textfield mdl-js-textfield">
      <input
        className="mdl-textfield__input"
        id="search-input"
        onChange={[Function]}
        type="text"
        value={undefined} />
      <label
        className="mdl-textfield__label"
        htmlFor="search-input">
        Enter Search Query
      </label>
    </div>
    <button
      className="mdl-button mdl-js-button mdl-button--raised mdl-js-ripple-effect mdl-button--accent"
      type="submit">
      Search
    </button>
  </form>
</div>
`;
